templates/aws-stack.yml

```diff
@@ -1614,6 +1634,129 @@ Resources:
       Subnet1RouteCheck: !If [ CreateVpcResources, !Ref Subnet1Routes, !Ref "AWS::NoValue" ]
       RouteDefaultCheck: !If [ CreateVpcResources, !Ref RouteDefault, !Ref "AWS::NoValue" ]
 
+  VpcEndpointsSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Condition: EnableVpcEndpoints
+    Properties:
+      GroupDescription: Security group for VPC endpoints
+      VpcId: !Ref Vpc
+      SecurityGroupIngress:
+        - IpProtocol: tcp
+          FromPort: 443
+          ToPort: 443
+          CidrIp: 10.0.0.0/16
```

Comment:
Might be over engineering but could use the `CidrBlock` attribute from the VPC itself to make sure it aligns e.g.,
```
CidrIp: !GetAtt Vpc.CidrBlock
```